prog/mutation.go

```diff
@@ -419,49 +467,124 @@ type mutationArgs struct {
 	target        *Target
 	args          []Arg
 	ctxes         []ArgCtx
+	priorities    []float64
 	ignoreSpecial bool
 }
 
+const (
+	maxPriority = float64(10)
+	minPriority = float64(1)
+)
+
 func (ma *mutationArgs) collectArg(arg Arg, ctx *ArgCtx) {
+	var prio float64
+	var canMutate bool
+
 	ignoreSpecial := ma.ignoreSpecial
 	ma.ignoreSpecial = false
 	switch typ := arg.Type().(type) {
-	case *StructType:
+	case *StructType, *UnionType:
```

Comment:
I think it's better to move this into corresponding `getMutationPrio()` functions after all, and pass e.g. `ma.target` as an argument. Otherwise this looks difficult to understand (for example I'm not sure if we need to keep that `len(typ.Fields) == 1` check in the `*UnionType` case here or not).

Reply:
Do you suggest to remove the entire switch or only the special type check? Also, if I will pass `ma.target` as argument I still have to check the `ignoreSpecial` flag. Can you elaborate more?

Reply:
Yes, remove this switch and have something like:

``` go
func (t *StructType) getMutationPrio(ma *mutationArgs) (prio float64, ctxStop bool, canMutate bool) {
	if ma.target.SpecialTypes[typ.Name()] == nil || ma.ignoreSpecial {
		return 0, false, false
	}
	return maxPriority, true, true
}
```

Reply:
The `mutation Args` object is created after the `chooseCall` function in which the `getMutationPrio()` functions are also used. Therefore we cannot use the `ma *mutationArgs` argument.

Reply:
I guess we can leave the switch as is then. @dvyukov WDYT?